### PR TITLE
fix(confirm-dialog): putting button computing in mapStateToProps

### PR DESCRIPTION
**Pull request by jlame**
**What is the problem this PR is trying to solve?**
The confirm dialog container was not updating when using expressions even if it was cmfconnected. The render function wasn't called when the props were unchanged which resulted in not evaluating expressions even when there was a change.

**What is the chosen solution to this problem?**
Computing validateAction and cancelAction in mapStateToProps in order to force a render when cmf stateToProps is called.
**Please check if the PR fulfills these requirements**
- [ ] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/containers/src/ConfirmDialog/ConfirmDialog.test.js [hunk 1]
```diff
@@ -51,6 +71,13 @@ describe('Connected ConfirmDialog', () => {
 	});
 });
 
+describe('Connected ConfirmDialog', () => {
```
Comment (jsomsanith) [inline packages/containers/src/ConfirmDialog/ConfirmDialog.test.js]: Remove that, you copied/pasted the block

File: packages/containers/src/ConfirmDialog/ConfirmDialog.test.js [hunk 2]
```diff
@@ -1,18 +1,36 @@
...
 
-jest.mock(
-	'@talend/react-components',
-	() => ({ ConfirmDialog: props => (<div className="tc-confirm-dialog" {...props} />) })
-);
+jest.mock('@talend/react-components', () => ({
+	ConfirmDialog: props => <div className="tc-confirm-dialog" {...props} />,
+}));
+
+describe('Action.mapStateToProps', () => {
```
Comment (jsomsanith) [inline packages/containers/src/ConfirmDialog/ConfirmDialog.test.js]: `Connected ConfirmDialog mapStateToProps `
Or better, move that into `describe('Connected ConfirmDialog')`

File: packages/containers/src/ConfirmDialog/ConfirmDialog.connect.js [hunk 3]
```diff
@@ -1,7 +1,26 @@
...
 
+export function mapStateToProps(state, props, cmfProps) {
+	const context = {
+		registry: api.registry.getRegistry(),
+		store: {
+			getState: () => state,
+		},
+	};
+	const validateAction = cmfProps.state ? cmfProps.state.getIn(['validateAction']) : undefined;
+	const cancelAction = cmfProps.state ? cmfProps.state.getIn(['cancelAction']) : undefined;
```
Comment (jmfrancois) [inline packages/containers/src/ConfirmDialog/ConfirmDialog.connect.js]:
```javascript
const validateAction = cmfProps.state ? cmfProps.state.getIn(['validateAction']) : cmfProps.validateAction;
const cancelAction = cmfProps.state ? cmfProps.state.getIn(['cancelAction']) : cmfProps.cancelAction;
```

File: packages/containers/src/ConfirmDialog/ConfirmDialog.connect.js [hunk 4]
```diff
@@ -1,7 +1,26 @@
...
+	const model = cmfProps.state ? cmfProps.state.getIn(['model']) : undefined;
```
Comment (jmfrancois) [inline packages/containers/src/ConfirmDialog/ConfirmDialog.connect.js]: same here

File: packages/containers/src/ConfirmDialog/ConfirmDialog.container.js [hunk 5]
```diff
@@ -30,12 +31,9 @@ class ConfirmDialog extends React.Component {
 		// and adding onClick event using the retrieved  action creator
 		state.validateAction = getActionsProps(this.context, state.validateAction, state.model);
 		state.cancelAction = getActionsProps(this.context, state.cancelAction, state.model);
+		const props = { ...omit(this.props, cmfConnect.INJECTED_PROPS), ...state };
```
Comment (jmfrancois) [inline packages/containers/src/ConfirmDialog/ConfirmDialog.container.js]: remove line 32 and 33
```javascript
const props = { ...omit(this.props, cmfConnect.INJECTED_PROPS), ...omit(state, ['validateAction', 'cancelAction']) };
```

File: packages/containers/src/ConfirmDialog/ConfirmDialog.connect.js (hunk 4 above)
Comment (jmfrancois) [reply packages/containers/src/ConfirmDialog/ConfirmDialog.connect.js]: replace with get and cmfProps.model

File: packages/containers/src/ConfirmDialog/ConfirmDialog.container.js (hunk 5 above)
Comment (jlame) [reply packages/containers/src/ConfirmDialog/ConfirmDialog.container.js]: as discussed, we're keeping them for now

**Comment by jmfrancois** [comment]
I approve this PR but a big re-work is needed:
Do not use getActionProps at all.
We should use the Inject mecanism with just cmfConnect is enough. So the action Id are resolved by the Action container.

**Pull request merged**